monash-human-power/dashboard
# Make battery voltage value persistent

## Description

Currently, the battery voltage for the camera systems and the WM are not retrieved upon browser refresh. This means you can only view the voltage if you're on the status page at the time the voltage is sent (which happens once every 5 mins currently). With this fix the voltage will be retrieved from storage in the backend to display it whenever you open the status page.

## Screenshots

<img width="490" alt="Screen Shot 2022-05-25 at 10 30 54 am" src="https://user-images.githubusercontent.com/63642262/170153888-c8dcd35c-3c13-473e-b35a-42a449c99df0.png">

Note: This is an old screenshot, the battery voltage should be rounded to 2 dp.
<img width="503" alt="Screen Shot 2022-05-18 at 3 11 41 pm" src="https://user-images.githubusercontent.com/63642262/170153899-284c4d71-2b50-4db2-a17b-f19513dda016.png">

## Steps to Test

1) Run dashboard, `yarn start` in client and server folders
2) publish the following to show the primary camera and WM 3 as online (you can however use any of the camera system and WM):
```
mosquitto_pub -t '/v3/wireless_module/3/status' -m '{"online": true}'
```
```
mosquitto_pub -t 'status/camera/primary' -m '{"connected": true,"ipAddress": "127.0.0.1","brightness": 4,"contrast": 3}'
```
3) Now publish the voltage
```
mosquitto_pub -t '/v3/wireless_module/3/battery' -m '{"voltage": 4.238497}'
```
```
mosquitto_pub -t 'camera/primary/battery' -m '{"voltage": 3}'
```
4) Refresh the page or switch to another page on dashboard and come back to status page.
5) Verify the battery voltage is still displayed.
